fixed checkbox valiidation error and alginment

vara-prasad-07 opened the pull request:
## Proposed Changes
What I changed in care_fe/src/components/Patient/PatientRegistration.tsx:
Fixes the address validation issue where validation messages were incorrectly displayed under the wrong field when the "Use this address for permanent address" checkbox was checked.
1. Fixed conditional validation logic: Updated the Zod schema to make permanent_address conditionally required only when same_address is unchecked, ensuring validation errors appear under the correct field.

2. Fixed form validation flow: Ensured that when the checkbox is checked, validation errors for empty addresses appear under the "Current Address" field instead of the "Permanent Address" field.

Fixes #13303 
The issue was in the validation schema where both address and permanent_address were always required. I implemented conditional validation using Zod's superRefine method:
1. When same_address is true: Only address is required (existing base schema handles this)
2. When same_address is false: permanent_address becomes required through custom validation
3. This ensures users see validation errors under the appropriate field based on their checkbox selection.

View here: 
<img width="824" height="386" alt="image" src="https://github.com/user-attachments/assets/425cce69-7475-49db-8942-17a196641cb0" />

<img width="753" height="140" alt="image" src="https://github.com/user-attachments/assets/9f592b08-784c-44ff-a4d4-0c89cabc0758" />


Tagging: @ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate the bug or test the new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is placed in I18n files.
- [x] Prepare a screenshot or demo video for the changelog entry and attach it to the issue.
- [ ] Request peer reviews.
- [ ] Complete QA on mobile devices.
- [ ] Complete QA on desktop devices.


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- Bug Fixes
  - Permanent address is required only when appropriate (non-minimal registration and not marked “same as current”), with clear field-specific errors.
  - Validation runs immediately when toggling “same as current address,” and permanent address auto-fills, clears, or disables consistently when copying from current address.

- Style
  - Registration layout adjusted: nationality now shows in two columns on medium screens; region fields stack in a single column for improved readability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Commit:
fixed checkbox valiidation error and alginment

vara-prasad-07 commented:
Hi @Jacobjeevan @rithviknishad ,
I messed up with previous pr so, i opened a new pr again.
I fixed the issue. Could any of you review this pr
Thank you

Denyme24 commented on src/components/Patient/PatientRegistration.tsx:
```diff
@@ -206,7 +204,18 @@ export default function PatientRegistration(
+              !data.permanent_address ||
+              data.permanent_address.trim() === ""
+            ) {
+              ctx.addIssue({
```
can we not put this custom validation inside `.superRefine((data, ctx) => { ... })`  to run extra validation checks after it finishes checking the basic rule.

vara-prasad-07 replied:
I will try to move custom validation logic for permanent address check inside the .superRefine() method, using ctx.addIssue() to report validation errors instead of handling them separately.

vara-prasad-07 replied:
@Denyme24 ,
This is exactly what .superRefine() is designed for - complex validations that depend on multiple fields or external conditions (like our minimalPatientRegistration flag).
 .superRefine() method gives us access to the entire data object and validation context, which is exactly what we need for this cross-field conditional validation. I would say this is the most appropriate solution for our use case.

nihal467 committed:
Merge branch 'develop' into checkboxissue

nihal467 commented:
@vara-prasad-07 once the PR is ready, tag me for review

Commit:
changes made to zod schema permenant address

Commit:
Merge branch 'checkboxissue' of https://github.com/vara-prasad-07/care_fe into checkboxissue

Commit:
made changes according to coderabbitai

Jacobjeevan commented on src/components/Patient/PatientRegistration.tsx:
```diff
@@ -132,9 +132,18 @@ export default function PatientRegistration(
+                message: t("field_required"),
+              });
+            }
+          }),
```
compare with existing logic, this bypasses minimalPatientRegistration

Jacobjeevan commented on src/components/Patient/PatientRegistration.tsx:
```diff
@@ -864,9 +872,26 @@ export default function PatientRegistration(
                                     "permanent_address",
-                                    form.getValues("address"),
-                                    { shouldValidate: true },
+                                    form.getValues("address") || "",
```
can simplify this block since lot of the code is same.

Jacobjeevan commented on src/components/Patient/PatientRegistration.tsx:
```diff
@@ -955,7 +984,7 @@ export default function PatientRegistration(
               </div>
 
-              <div className="grid grid-cols-1 md:grid-cols-2 gap-4">
+              <div className="grid grid-cols-1 gap-4">
```
don't change this, rm'ing results in user having to scroll more.

Jacobjeevan reviewed (changes requested):
Always review your changes against existing functionality.

Commit:
Merge branch 'develop' into checkboxissue

Commit:
made changes according to existing logic

vara-prasad-07 commented:
Hi @Jacobjeevan 
I made changes based on your recent comments

1. Fixed! Updated the permanent_address validation to respect the minimalPatientRegistration flag. Now it's optional when minimalPatientRegistration is true, and uses the existing validation logic when false 

2. Simplified the checkbox logic by consolidating the duplicate if-else blocks into cleaner code with variables. Reduced from ~15 lines to ~10 lines while maintaining the same functionality.

3. Reverted the grid layout change back to `grid-cols-1 md:grid-cols-2` to maintain the responsive two-column layout and prevent unnecessary scrolling.

working video:

https://github.com/user-attachments/assets/470bb1c4-7182-497c-a2eb-0cf6ee206730

Thank you
